jbmc/src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -708,12 +708,14 @@ void java_bytecode_convert_classt::convert(
       new_symbol.type.set(ID_C_access, ID_default);
 
     const namespacet ns(symbol_table);
-    new_symbol.value=
-      zero_initializer(
-        field_type,
-        class_symbol.location,
-        ns,
-        get_message_handler());
+    const auto value = zero_initializer(field_type, class_symbol.location, ns);
+    if(!value.has_value())
+    {
+      error().source_location = class_symbol.location;
+      error() << "failed to zero-initialize " << f.name << eom;
+      throw 0;
```

Comment:
:bulb: Throwing an int isn't great - would it make sense to introduce a structured exception here?

Reply:
Ack, but three different locations in that file currently follow three different approaches. +1 for cleaning this up, but possibly out of scope for this PR?